pkg/queue/health/probe.go

```diff
@@ -17,18 +17,88 @@ limitations under the License.
 package health
 
 import (
+	"crypto/tls"
+	"errors"
+	"fmt"
 	"net"
+	"net/http"
+	"net/url"
 	"time"
+
+	"github.com/knative/serving/pkg/network"
+	"github.com/knative/serving/pkg/queue"
+	corev1 "k8s.io/api/core/v1"
 )
 
+// HTTPProbeConfigOptions holds the HTTP probe config options
+type HTTPProbeConfigOptions struct {
+	Timeout time.Duration
+	*corev1.HTTPGetAction
+}
+
+// TCPProbeConfigOptions holds the TCP probe config options
+type TCPProbeConfigOptions struct {
+	SocketTimeout time.Duration
+	Address       string
+}
+
 // TCPProbe checks that a TCP socket to the address can be opened.
 // Did not reuse k8s.io/kubernetes/pkg/probe/tcp to not create a dependency
 // on klog.
-func TCPProbe(addr string, socketTimeout time.Duration) error {
-	conn, err := net.DialTimeout("tcp", addr, socketTimeout)
+func TCPProbe(config TCPProbeConfigOptions) error {
+	conn, err := net.DialTimeout("tcp", config.Address, config.SocketTimeout)
 	if err != nil {
 		return err
 	}
 	conn.Close()
 	return nil
 }
+
+// HTTPProbe checks that HTTP connection can be established to the address.
+func HTTPProbe(config HTTPProbeConfigOptions) error {
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			DisableKeepAlives: true,
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+		Timeout: config.Timeout,
+	}
+	url := url.URL{
+		Scheme: string(config.Scheme),
+		Host:   config.Host + ":" + config.Port.String(),
+		Path:   config.Path,
+	}
+	req, err := http.NewRequest(http.MethodGet, url.String(), nil)
+	if err != nil {
+		return fmt.Errorf("error constructing probe request %v", err)
+	}
+
+	req.Header.Add(network.KubeletProbeHeaderName, queue.Name)
```

Comment:
This should pass this check: https://github.com/knative/serving/blob/bc76efe828840fcabd8df1bce3b7c5be065489f2/pkg/network/network.go#L323-L324